src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,99 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double OVERLOADING_FACTOR = 0.75;
+    private static final int INITIAL_SIZE = 16;
+    private int size;
+    private Node<K, V>[] array = new Node[INITIAL_SIZE];
+
+    private static class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value) {
+            this.hash = (key == null ? 0 : key.hashCode());
+            if (hash < 0) {
+                hash = hash * -1;
+            }
```

Comment:
you should pass in the constructor already calculated hash, also, it's a bad practice to execute such logic inside of the constructor, the only function of the constructor is an initialization of fields